network/p2p/client.go

```diff
@@ -42,11 +42,11 @@ type CrossChainAppResponseCallback func(
 )
 
 type Client struct {
-	self          ids.NodeID
 	handlerID     uint64
 	handlerIDStr  string
 	handlerPrefix []byte
 	router        *router
+	peers         *Peers
```

Comment:
I was trying to avoid directly referencing the `Peers` struct in the client to support custom samplers (like sampling validators by stake). I feel like defining an interface we're happy with here is how we should be enabling different sampling mechanisms.

How would this look if we were enabling sampling by stake?

Reply:
Added an interface that exposes the peer slice